Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CalculateMonteCarloAbsorption.py

```diff
@@ -0,0 +1,794 @@
+                ss_monte_carlo_alg.setProperty("InputWorkspace", container_wave_2)
+                ss_monte_carlo_alg.setProperty("Thickness", self._container_back_thickness)
+                ss_monte_carlo_alg.setProperty("Center", offset_back)
+                ss_monte_carlo_alg.execute()
+                acc_2 = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+                acc_ws = self._multiply(acc_1, acc_2)
+
+            elif self._shape == 'Cylinder':
+                ss_monte_carlo_alg.setProperty("InputWorkspace", container_wave_1)
+                ss_monte_carlo_alg.setProperty("InnerRadius", self._container_inner_radius)
+                ss_monte_carlo_alg.setProperty("OuterRadius", self._container_outer_radius)
+                ss_monte_carlo_alg.setProperty("Shape", "Annulus")
+                ss_monte_carlo_alg.execute()
+                acc_ws = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+            elif self._shape == 'Annulus':
+                ss_monte_carlo_alg.setProperty("InputWorkspace", container_wave_1)
+                ss_monte_carlo_alg.setProperty("InnerRadius", self._container_inner_radius)
+                ss_monte_carlo_alg.setProperty("OuterRadius", self._container_outer_radius)
+                ss_monte_carlo_alg.execute()
+                acc_1 = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+                ss_monte_carlo_alg.setProperty("InputWorkspace", container_wave_2)
+                ss_monte_carlo_alg.execute()
+                acc_2 = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+                acc_ws = self._multiply(acc_1, acc_2)
+
+            for log_name, log_value in container_kwargs.items():
+                sample_log_names.append("container_" + log_name.lower())
+                sample_log_values.append(log_value)
+
+            acc_ws = self._convert_from_wavelength(acc_ws)
+
+            self._add_sample_log_multiple(acc_ws, sample_log_names, sample_log_values)
+
+            if not self.isChild():
+                mtd.addOrReplace(self._acc_ws_name, acc_ws)
+
+            self._output_ws = self._group_ws([ass_ws, acc_ws])
+        else:
+            self._output_ws = self._group_ws([ass_ws])
+
+        self.setProperty('CorrectionsWorkspace', self._output_ws)
+
+    def _setup(self):
+
+        # The beam properties and monte carlo properties are simply passed straight on to the
+        # SimpleShapeMonteCarloAbsorptionCorrection algorithm so they are being put into
+        # a dictionary for simplicity
+        sample_workspace = self.getProperty("SampleWorkspace").value
+        container_workspace = self.getProperty("ContainerWorkspace").value
+        sample_is_group = isinstance(sample_workspace, WorkspaceGroup)
+        container_is_group = isinstance(container_workspace, WorkspaceGroup)
+
+        if container_workspace and sample_is_group != container_is_group:
```

Comment:
Not critical, but these condition blocks read hard, any way to streamline better?